File: LeetCode/108. 将有序数组转换为二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sortedArrayToBST = function (nums) {
  if (!nums.length) {
    return null
  }
  const index = Math.floor(nums.length / 2)
  const node = new TreeNode(nums[index])
  node.left = sortedArrayToBST(nums.slice(0, index))
  node.right = sortedArrayToBST(nums.slice(index + 1))
  return node
}

File: LeetCode/101-200/108. 将有序数组转换为二叉搜索树.java
    /**
    * Definition for a binary tree node.
    * public class TreeNode {
    *     int val;
    *     TreeNode left;
    *     TreeNode right;
    *     TreeNode() {}
    *     TreeNode(int val) { this.val = val; }
    *     TreeNode(int val, TreeNode left, TreeNode right) {
    *         this.val = val;
    *         this.left = left;
    *         this.right = right;
    *     }
    * }
    */
    class Solution {
        public TreeNode sortedArrayToBST(int[] nums) {
         return   help(nums, 0, nums.length -1);
        }
         public  TreeNode help(int[] nums, int left, int right) {
            if (left > right) return null;
            int index = (left + right) / 2;
            TreeNode node = new TreeNode(nums[index]);
            node.left = help(nums, left, index - 1);
            node.right = help(nums, index + 1,right);
            return  node;
    }
    }